lib/util/node-event-generator.js

```diff
@@ -160,7 +160,7 @@ function tryParseSelector(rawSelector) {
         return esquery.parse(rawSelector.replace(/:exit$/, ""));
     } catch (err) {
         if (typeof err.offset === "number") {
-            throw new Error(`Syntax error in selector "${rawSelector}" at position ${err.offset}: ${err.message}`);
+            throw new TypeError(`Syntax error in selector "${rawSelector}" at position ${err.offset}: ${err.message}`);
```

Comment:
Is a `TypeError` semantically correct here (as opposed to, e.g. a `SyntaxError`)? This error doesn't seem to relate to types.